File: Placement Sheet/Tree/CheckBST.ts
class TreeNode {
    private data: number;
    private left: TreeNode | null;
    private right: TreeNode | null;

    constructor(data: number) {
        this.data = data;
        this.left = null;
        this.right = null;
    }

    getData(): number {
        return this.data;
    }

    getLeft(): TreeNode | null {
        return this.left;
    }

    getRight(): TreeNode | null {
        return this.right;
    }

    setLeft(node: TreeNode): void {
        this.left = node;
    }

    setRight(node: TreeNode): void {
        this.right = node;
    }
}

function isBST(node: TreeNode | null, min: number = -Infinity, max: number = Infinity): boolean {
    if (!node) {
        return true;
    }
    const val: number = node.getData();

    if (val <= min || val >= max) {
        return false;
    }
    return isBST(node.getLeft(), min, val) && isBST(node.getRight(), val, max);
}

let root: TreeNode = new TreeNode(10);
let left: TreeNode = new TreeNode(5);
let right: TreeNode = new TreeNode(15);

root.setLeft(left);
root.setRight(right);

console.log(isBST(root));

export { };

File: Placement Sheet/Tree/CheckBST.java
public class CheckBST {
    public static class TreeNode {
        private int data;
        private TreeNode left;
        private TreeNode right;

        TreeNode(int data) {
            this.data = data;
            this.left = null;
            this.right = null;
        }

        public int getData() {
            return data;
        }

        public TreeNode getLeft() {
            return left;
        }

        public TreeNode getRight() {
            return right;
        }

        public void setRight(TreeNode right) {
            this.right = right;
        }

        public void setLeft(TreeNode left) {
            this.left = left;
        }
    }

    public static boolean isBST(TreeNode node, int min, int max) {
        if (node == null) {
            return true;
        }

        int val = node.getData();

        if (val <= min || val >= max) {
            return false;
        }

        return isBST(node.getLeft(), min, val) && isBST(node.getRight(), val, max);
    }

    public static void main(String[] args) {
        TreeNode root = new TreeNode(10);
        TreeNode left = new TreeNode(5);
        TreeNode right = new TreeNode(15);
        root.setLeft(left);
        root.setRight(right);
        System.out.println(isBST(root, Integer.MIN_VALUE, Integer.MAX_VALUE));
    }
}
